TypeCobol.LanguageServer/WorkspaceProjectStore.cs

```diff
@@ -0,0 +1,286 @@
+        /// <exception cref="DuplicatedProjectException">If a project with the same key already exists in the store</exception>
+        internal void AddProject(WorkspaceProject project)
+        {
+            System.Diagnostics.Debug.Assert(project != null && project.ProjectKey != null);
+            if (!_workspaceProjects.TryAdd(project.ProjectKey, project))
+            {
+                throw new DuplicatedProjectException(project.ProjectKey);
+            }
+        }
+
+        /// <summary>
+        /// Create a WorkspaceProject instance, and add it to the store.
+        /// </summary>
+        /// <param name="projectKey">The Project's key</param>
+        /// <param name="configuration">The configuration to be applied</param>
+        /// <returns>The WorkspaceProject instance created</returns>
+        /// <exception cref="DuplicatedProjectException">Sent if the given Project's key is already associted to a project.</exception>
+        internal WorkspaceProject CreateWorkspaceProject(string projectKey, TypeCobolConfiguration configuration)
+        {
+            System.Diagnostics.Debug.Assert(projectKey != null);
+            if (_workspaceProjects.ContainsKey(projectKey))
+            {
+                throw new DuplicatedProjectException(projectKey);
+            }
+
+            CompilationProject compilationProject = new CompilationProject(projectKey, this._rootDirectoryFullName, Helpers.DEFAULT_EXTENSIONS,
+                configuration.Format, this.DefaultWorkspaceProject.Project.CompilationOptions, 
+                this.DefaultWorkspaceProject.Project.AnalyzerProvider);
+            WorkspaceProject workspaceProject = new WorkspaceProject(projectKey, compilationProject);
+            AddProject(workspaceProject);
+            return workspaceProject;
+        }
+
+        /// <summary>
+        /// Try to get the DocumentContext and the corresponding WokspaceProject instance associated to the given Uri
+        /// </summary>
+        /// <param name="fileUri">The File Uri</param>
+        /// <param name="openedDocumentContext"></param>
+        /// <param name="workspaceProject"></param>
+        /// <returns>true if the DocumentContext and the corresponding WokspaceProject instance have been found, false otherwise</returns>
+        internal bool TryGetOpenedWorkspaceDocumentProjet(Uri fileUri, out DocumentContext openedDocumentContext, out WorkspaceProject workspaceProject)
+        {
+            openedDocumentContext = null;
+            workspaceProject = null;
+            foreach (var wksProject in this)
+            {
+                if (wksProject.TryGetOpenedDocumentContext(fileUri, out openedDocumentContext))
+                {
+                    workspaceProject = wksProject;
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Try to get the DocumentContext and the corresponding WokspaceProject instance by name
```

Comment:
typo: 'WorkspaceProject'

Reply:
Changed in [e1c296c](https://github.com/TypeCobolTeam/TypeCobol/pull/2145/commits/e1c296c5835196790db878e8efdab69f6a196c18)